From 5de1bb82465d39962e26175c62f644a3e423d030 Mon Sep 17 00:00:00 2001
From: Adam Williamson <awilliam@redhat.com>
Date: Thu, 21 Jan 2016 17:21:52 -0800
Subject: [PATCH] use 'exec' in py3-compatible manner

per https://docs.python.org/2/reference/simple_stmts.html ,
as exec is a function not a statement in py3, the py2 version
has been set to allow the subsequent statement to be a tuple,
so we can invoke it like this to make it both py2 and py3
compatible. Without this, byte-compiling the file fails under
py3.

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
downloaded from upstream commit:
https://github.com/mwclient/mwclient/commit/5de1bb82465d39962e26175c62f644a3e423d030

 mwclient/ex.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mwclient/ex.py b/mwclient/ex.py
index db4006c..c0b1eae 100644
--- a/mwclient/ex.py
+++ b/mwclient/ex.py
@@ -12,7 +12,7 @@ def read_config(config_files, **predata):
 
 def _read_config_file(_config_file, predata):
     _file = open(_config_file)
-    exec _file in globals(), predata
+    exec(_file, globals(), predata)
     _file.close()
 
     for _k, _v in predata.iteritems():
-- 
2.9.3

